index.d.ts

```diff
@@ -0,0 +1,438 @@
+import {
+  UnwrapComputedPropertyGetter,
+  UnwrapComputedPropertyGetters
+} from '@ember/object/-private/types';
+
+import RSVP from 'rsvp';
+
+// Lifted from @types/ember__object/observable.d.ts
+interface Getter {
+  /**
+   * Retrieves the value of a property from the object.
+   */
+  get<K extends keyof this>(key: K): UnwrapComputedPropertyGetter<this[K]>;
+  /**
+   * To get the values of multiple properties at once, call `getProperties`
+   * with a list of strings or an array:
+   */
+  getProperties<K extends keyof this>(
+    list: K[]
+  ): Pick<UnwrapComputedPropertyGetters<this>, K>;
+  getProperties<K extends keyof this>(
+    ...list: K[]
+  ): Pick<UnwrapComputedPropertyGetters<this>, K>;
+}
+
+export type GeneratorFn<T, Args extends any[] = any[], R = any> = (
+  this: T,
+  ...args: Args
+) => IterableIterator<R>;
+
+export const all: typeof Promise.all;
+export const allSettled: typeof RSVP.allSettled;
+export const hash: typeof RSVP.hash;
+export const race: typeof Promise.race;
+
+export function timeout(ms: number): Promise<void>;
+
+/**
+ * Use `waitForEvent` to pause the task until an event is fired.
+ * The event can either be a jQuery event or an `Ember.Evented` event, or any
+ * event system where the object supports `.on()`, `.one()` and `.off()`.
+ */
+export function waitForEvent(
+  object: EmberObject | EventTarget,
+  eventName: string
+): Promise<Event>;
+
+/**
+ * Use `waitForProperty` to pause the task until a property on an object
+ * changes to some expected value.
+ *
+ * This can be used for a variety of use cases, including synchronizing with
+ * another task by waiting for it to become idle, or change state in some
+ * other way.
+ *
+ * If you omit the callback, `waitForProperty` will resume execution when the
+ * observed property becomes truthy. If you provide a callback, it'll be
+ * called immediately with the observed property's current value, and multiple
+ * times thereafter whenever the property changes, until you return a truthy
+ * value from the callback, or the current task is canceled.
+ *
+ * You can also pass in a non-`function` value in place of the callback, in
+ * which case the task will continue executing when the property's value
+ * becomes the value that you passed in.
+ */
+export function waitForProperty<T extends object, K extends keyof T>(
+  object: T,
+  key: K,
+  callbackOrValue?: (value: T[K]) => boolean | any
+): Promise<void>;
+
+/**
+ * Use `waitForQueue` to pause the task until a certain run loop queue is
+ * reached.
+ */
+export function waitForQueue(queueName: string): Promise<void>;
+
+export function task<Args extends any[], R>(
+  taskFn: GeneratorFn<unknown, Args, R>
+): Task<Args, Exclude<R, Promise<any>>>;
+export function task<Args extends any[], R>(encapsulatedTask: {
+  perform: GeneratorFn<unknown, Args, R>;
+}): Task<Args, Exclude<R, Promise<any>>>;
+
+export function taskGroup(): TaskGroupProperty;
+
+interface CommonTaskProperty {
```

Comment:
should this also have `.withTestWaiter`? or is it recommended that people use the wrapping version?